vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -48,7 +63,78 @@ public void setExhibition(JoinPoint jp) {
+    public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+        Object[] args = jp.getArgs();
+        MethodSignature signature = (MethodSignature) jp.getSignature();
+        int indexOfModel = (Arrays.asList(signature.getParameterTypes())).indexOf(Model.class);
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition(); 
+        // If there is no exhibition, we go back to root url page.
+        if(exhibition==null) {
+            return "redirect:/";
+        }
+        //If no exhibition mode has been setup for existing exhibition, we skip modes and aspects.
+        if(exhibition.getMode() == null) {
+            return jp.proceed();
+        }
+        Map<String, String> ids = getIds(args, signature);
+        String spaceId = ids.getOrDefault(IdPrefix.SPACEID.getValue(), "");
+        String moduleId = ids.getOrDefault(IdPrefix.MODULEID.getValue(), "");
+        return redirectRequest(jp, spaceId, moduleId, indexOfModel, exhibition);
+    }
+    
+    // Based on exhibition mode, get the redirect page or pass control to controller.  
+    public Object redirectRequest(ProceedingJoinPoint jp, String spaceId, String moduleId, int modelIndex, Exhibition exhibition) throws Throwable{
+        ISpace space = spaceManager.getSpace(spaceId);
+        IModule module = moduleManager.getModule(moduleId);
+        Object[] args = jp.getArgs();
+        // For exhibition being active, if space or module is not found, show not found page or else give control to controller.
+        if(exhibition.getMode().equals(ExhibitionModes.ACTIVE)) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            return jp.proceed();
+        }
+        // If the user is logged in, if space or module invalid, show not found page or else add attribute to show pop up
+        // notifying user about status of exhibition. ModelIndex checks if there is a model to add attribute to.
+        if(authFacade.getAuthenticatedUser()!=null && modelIndex>-1) {
+            if(space==null && module==null) {
+                return "notFound";
+            }
+            ((Model) args[modelIndex]).addAttribute("showModal", "true"); 
+            return jp.proceed();
+        }
+        // When exhibition is offline, if custom message is setup, show that to user, or else show default message, or else
+        // show maintenance default message if exhibition is set to maintenance mode.
+        if(exhibition.getMode()==ExhibitionModes.OFFLINE) {
+            String modeValue = exhibition.getCustomMessage().equals("") == false ? exhibition.getCustomMessage() : exhibition.getMode().getValue();
+            ((Model) args[modelIndex]).addAttribute("modeValue", modeValue);
+        } else {
+            ((Model) args[modelIndex]).addAttribute("modeValue", exhibition.getMode().getValue());
+        }
+        return "maintenance";
```

Comment:
this is still difficult to read and unnecessary complex. Try this:

- check if mode is offline or maintenance (and no logged in user) if so set message and return maintenance
- check if space or module are null, if so return null
- check if user is logged in and offline or maintenance if so, show message
- otherwise proceed

you probably don't even need the mode==null check as this would be handled by the above points